### ISSUE TITLE
Unable to attach existing node to a pool in version 1.4.0

### ISSUE THREAD

**Issue by chihaiaalex** [F5Networks/terraform-provider-bigip#380]
### Environment

 * TMOS/Bigip Version: BIG-IP 12.1.2 Build 2.0.276 Hotfix HF2
 * Terraform Version: v0.13.5
 * Terraform bigip provider Version: 1.4.0

### Summary

With the new version 1.4.0 of the BigIP provider, I am unable to atach existing nodes to a pool.

### Steps To Reproduce

1. Provide terraform resource config which you are facing trouble along with the output of it.

```
resource "bigip_ltm_pool" "k8s_api_pool" {
  name = "/k8s/${var.cluster.k8s_api.pool}"
  load_balancing_mode = "round-robin"
  description = "Pool for k8s api"
  monitors = [
    "/Common/tcp"
  ]
  allow_snat = "yes"
  allow_nat = "yes"
}
resource "bigip_ltm_node" "nodes" {
  for_each = {for node in local.all_nodes : node.fqdn => node}
  name = "/k8s/${each.value.fqdn}"
  address = "${each.value.ip}%52"
}
resource "bigip_ltm_pool_attachment" "k8s_api_pool_attachment" {
  for_each = {for node in var.cluster.master_nodes: node.fqdn => node}
  pool = bigip_ltm_pool.k8s_api_pool.name
  node = "/k8s/${each.value.fqdn}:6443"
}
```

Same code works perfectly in version 1.3.2 of the provider.

I also tried this kind of reference:

```
resource "bigip_ltm_pool_attachment" "k8s_api_pool_attachment" {
  for_each = {for node in var.cluster.master_nodes: node.fqdn => node}
  pool = bigip_ltm_pool.k8s_api_pool.name
  node = "${bigip_ltm_node.nodes[each.value.fqdn]}:6443"
}
```

### Error is: 

```
Error: Failure adding node /k8s/my-fqdn.example.com:6443 to pool /k8s/k8sapi_TCP6443_POOL: 01070734:3: Configuration error: Node (/k8s/my-fqdn.example.com) already exists as a IP (nn.nn.nn.nnn%52)
  on modules\bigip\main.tf line 41, in resource "bigip_ltm_pool_attachment" "k8s_api_pool_attachment":
  41: resource "bigip_ltm_pool_attachment" "k8s_api_pool_attachment" {
```

### Expected Behavior

In the docs says that: "There are two ways to use ltm_pool_attachment resource, where we can take node reference from ltm_node or we can specify node directly with ip:port/fqdn:port which will also create node and atach to pool.".

So I would expect to be able to reference an existing node and attach it to a pool, given that the same nodes are being used in many other pools.

**Comment by chihaiaalex** [comment]
Investigating further, I realized that chaning the node name from a fqdn (my-fqdn.example.com) to my-fqdn will make it work. This is so strange.

**Comment by nmenant** [comment]
Hi,

i've been able to reproduce your issue: 

``` 
resource "bigip_ltm_pool" "pool" {
  name                = "/test/terraform-test-pool"
  load_balancing_mode = "round-robin"
  allow_snat          = "yes"
  allow_nat           = "yes"
}
resource "bigip_ltm_node" "node" {
  name    = "/test/www.perdu.com"
  address = "208.97.177.124"
}

resource "bigip_ltm_pool_attachment" "attach_node" {
  pool = bigip_ltm_pool.pool.name
  node = "/test/www.perdu.com:6443"
}

PAR-ML-00026375:github380 menant$ terraform apply --auto-approve
bigip_ltm_node.node: Creating...
bigip_ltm_pool.pool: Creating...
bigip_ltm_node.node: Creation complete after 0s [id=/test/www.perdu.com]
bigip_ltm_pool.pool: Creation complete after 0s [id=/test/terraform-test-pool]
bigip_ltm_pool_attachment.attach_node: Creating...
bigip_ltm_pool_attachment.attach_node: Creation complete after 0s [id=/test/terraform-test-pool-/test/www.perdu.com:6443]
``` 

it creates all the right config: 
![image](https://user-images.githubusercontent.com/13098758/99096368-7882d000-25d6-11eb-9bcc-8ad89eb4a1a2.png)


![image](https://user-images.githubusercontent.com/13098758/99096407-85072880-25d6-11eb-89b2-ded1215bb368.png)


When switching to v1.4.0 i also have an error message (although different than yours)

``` 
PAR-ML-00026375:github380 menant$ terraform init

Initializing the backend...

Initializing provider plugins...
- Using previously-installed f5networks/bigip v1.4.0

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
PAR-ML-00026375:github380 menant$ terraform apply --auto-approve
bigip_ltm_pool.pool: Creating...
bigip_ltm_node.node: Creating...
bigip_ltm_node.node: Creation complete after 0s [id=/test/www.perdu.com]
bigip_ltm_pool.pool: Creation complete after 0s [id=/test/terraform-test-pool]
bigip_ltm_pool_attachment.attach_node: Creating...

Error: Failure adding node /test/www.perdu.com:6443 to pool /test/terraform-test-pool: HTTP 400 :: {"code":400,"message":"01070734:3: Configuration error: Node (/test/www.perdu.com) is configured using static IP (208.97.177.124). FQDN property cannot be used with static IP nodes. Delete the node if you want to reconfigure it as FQDN.","errorStack":[],"apiError":3}

  on main.tf line 19, in resource "bigip_ltm_pool_attachment" "attach_node":
  19: resource "bigip_ltm_pool_attachment" "attach_node" {

``` 

I can get the same error message when trying to add a pool member using FQDN name resolution
![image](https://user-images.githubusercontent.com/13098758/99096954-3443ff80-25d7-11eb-9f56-5dd9bba2400b.png)


My take is the following: 
```
resource "bigip_ltm_pool_attachment" "attach_node" {
  pool = bigip_ltm_pool.pool.name
  node = "/test/www.perdu.com:6443"
}
```
v1.3.2: This is translated as "assign the following node as a pool member"

v1.4.0: This is translated as "do a name resolution and add this as a pool member"

CREATED TER-506 to investigate this

**Comment by papineni87** [comment]
@chihaiaalex can you pull the latest code from master branch and check if the issue persists ? i fixed regex for pool attachment which will fix the issue as well .I do verified the fix with above config and works for me.

**Comment by chihaiaalex** [comment]
@papineni87 I can confirm that with the last version from master is working as expected. Thanks!

**Comment by papineni87** [comment]
Thanks @chihaiaalex  for confirming the fix. We will push the fix in the coming release v1.5.0

**Comment by chihaiaalex** [comment]
@papineni87 The update of a pool member is also broken when the member was attached using an existing node. 

Example of change:
```
~ resource "bigip_ltm_pool_attachment" "aggregation_pool" {
        id             = "/k8s/k8singress-agg_TCP31380_POOL-/k8s/my-fqdn.example.com:31380"
        node           = "/k8s/my-fqdn.example.com:31380"
        pool           = "/k8s/k8singress-agg_TCP31380_POOL"
      ~ priority_group = 2 -> 3
    }
```
I get error: 
```
Error: Failure adding node /k8s/my-fqdn.example.com:31380 to pool /k8s/k8singress-agg_TCP31380_POOL-/k8s/my-fqdn.example.com:31380: 01020036:3: The requested Pool Member (/k8s/k8singress-agg_TCP31380_POOL-/k8s/my-fqdn.example.com:31380 /k8s/k8s/my-fqdn.example.com 31380) was not found.
```

Looking into code I see that the update functionality is expecting the id of the resource to represent the pool name, but that's not the case when the member is attached from an existing node. If you look at the create functionality for existing nodes, you can see that the id is differently set in the two cases:

When attaching a member from an existing node: `d.SetId(fmt.Sprintf("%s-%s", poolName, nodeName))`
When attaching a memeber from a newly created node: `d.SetId(poolName)`

Also, the fact that these two scenarios are treated differently causes another issue which was reported here #381 . When using an existing node the function `resourceBigipLtmPoolAttachmentUpdate` is not called and the priority group is not set in BigIP.

If I can contribute to these issues, please let me know. I would gladly do it.

**Comment by papineni87** [comment]
@chihaiaalex sorry for the delay,   yes as you said we are handling two scenerios differently to handle old and new behaviour of pool_attachment. I guess pool_priority for pool_attachment is broken which is causing this issue. As part of fix for #381 , this issue should also be tested and fixed. 

Yes, we will accept external pr's and that would be great for contribution of issues.

**Comment by RavinderReddyF5** [comment]
fixed
